src/main/java/net/spy/memcached/ArcusClient.java

```diff
@@ -2146,50 +2146,29 @@ public SMGetFuture<List<SMGetElement<Object>>> asyncBopSortMergeGet(
    * @return list of grouped (memcached node + keys) in the group
    */
   private Collection<Entry<MemcachedNode, List<String>>> groupingKeys(List<String> keyList, int groupSize) {
-    Map<String, Integer> chunkCount = new HashMap<String, Integer>();
-    Map<String, Entry<MemcachedNode, List<String>>> result = new HashMap<String,
-        Entry<MemcachedNode, List<String>>>();
-
+    List<Entry<MemcachedNode, List<String>>> resultList = new ArrayList<Entry<MemcachedNode, List<String>>>();
+    Map<MemcachedNode, List<String>> temp = new HashMap<MemcachedNode, List<String>>();
     MemcachedConnection conn = getMemcachedConnection();
 
-    for (String k : keyList) {
-      MemcachedNode qa = conn.findNodeByKey(k);
-      String node;
+    for (String key : keyList) {
+      MemcachedNode qa = conn.findNodeByKey(key);
+
       if (qa == null) {
-        node = "fake_node";
-      } else {
-        node = qa.getSocketAddress().toString();
+        throw new IllegalStateException("Could not find node for key: " + key);
       }
-      int cc;
-      if (chunkCount.containsKey(node)) {
-        cc = chunkCount.get(node);
-      } else {
-        cc = 0;
-        chunkCount.put(node, 0);
+      if (!temp.containsKey(qa)) {
+        temp.put(qa, new ArrayList<String>());
+      } else if (temp.get(qa).size() == groupSize) {
+        resultList.add(new AbstractMap.SimpleEntry<MemcachedNode, List<String>>(qa, temp.get(qa)));
+        temp.put(qa, new ArrayList<String>());
       }
-
-      String resultKey = node + cc;
-
-      List<String> arrangedKeyList = null;
-
-      if (result.containsKey(resultKey)) {
-        if (result.get(resultKey).getValue().size() >= groupSize) {
-          arrangedKeyList = new ArrayList<String>();
-          cc++;
-          result.put(node + cc, new AbstractMap.SimpleEntry<MemcachedNode,
-              List<String>>(qa, arrangedKeyList));
-          chunkCount.put(node, cc);
-        } else {
-          arrangedKeyList = result.get(resultKey).getValue();
-        }
-      } else {
-        arrangedKeyList = new ArrayList<String>();
-        result.put(resultKey, new AbstractMap.SimpleEntry<MemcachedNode,
-            List<String>>(qa, arrangedKeyList));
-      }
-      arrangedKeyList.add(k);
+      temp.get(qa).add(key);
     }
-    return result.values();
+    /*
+    * Add the Entry instance which is not full(smaller than groupSize) to the result.
+    * */

```

Comment:
```suggestion
    /**
     * Add the Entry instance which is not full(smaller than groupSize) to the result.
     */
```
또는 single line comment(`//`)로 충분하지 않은지?

Reply:
반영완료 하였습니다~